ast/src/analyzed/mod.rs

```diff
@@ -181,6 +181,18 @@ impl<T> Analyzed<T> {
                                     },
                                 );
                             }
+                            // we still need a replacement for zero sized array,
```

Comment:
I think we should not allocate IDs for zero-sized arrays. Maybe we could make it an error in `statement_processor` and/or in the condenser?

Reply:
it's already not allocated, but the Symbol is assigned the id where element id's would start from